### DISCUSSION TITLE
where can i find  the souce code of ImageUtils.java

### DISCUSSION THREAD

**Discussion by xuhongxin** [deepjavalibrary/djl#2923] [Development]
where can i find  the souce code of ImageUtils.java

**Comment by zachgk** [comment]
We don't have an `ImageUtils.java` right now. I think we had one in older versions of DJL as of several years ago. Where did you see the reference to it?

**Reply by xuhongxin** [reply]
![image](https://github.com/deepjavalibrary/djl/assets/1044109/7ffabd80-4884-4c8f-b8d7-cc2943f5540e)

https://d2l.djl.ai/chapter_linear-networks/image-classification-dataset.html

here 👆🏻

**Comment by frankfliu** [comment]
@xuhongxin 

Do you mean: https://github.com/deepjavalibrary/djl/blob/master/api/src/main/java/ai/djl/modality/cv/util/NDImageUtils.java?

**Reply by xuhongxin** [reply]
i was study chapter 3.5；and the demo use ImageUtils.showImages

![image](https://github.com/deepjavalibrary/djl/assets/1044109/242ca94d-5e7a-44e8-a82d-ddeb5a0e47ed)

**Comment by frankfliu** [comment]
That's utility class for Jupyter notebook. You can find the source code here: https://github.com/deepjavalibrary/d2l-java/blob/master/utils/ImageUtils.java
